Further update models

sarakonradi opened the pull request:
This PR is based on #1374 and further updates the model files.

It: 
- Ensures keywords are consistently written in singular form
- Writes out abbreviations (e.g. `AC` and `DC`)
- Updates the `Device name`
- Adds an empty line between `EndUserDocs` and the class definition
- Corrects model names, such as `iaf_psc_exp_htum`

sarakonradi committed:
Change keywords to singular form

sarakonradi committed:
Add empty line between EndUserDocs and class definition

sarakonradi committed:
Update keywords in BeginUserDocs

sarakonradi committed:
Correct iaf_psc_exp_htum model name

sarakonradi committed:
Write out AC abbreviation

sarakonradi committed:
Add device name to devices

sarakonradi committed:
Add description to devices

sarakonradi committed:
Add model name and dash

sarakonradi committed:
Remove synopsis

sarakonradi committed:
Write out DC abbreviation

sarakonradi committed:
Reformulate definition

sarakonradi committed:
Replace Transmits with Sends

jougs commented on models/ac_generator.h:
```diff
@@ -46,7 +46,7 @@ ac_generator
 +++++++++++
 
-This device produces an ac-current sent by a CurrentEvent. The
+This device produces an AC current sent by a CurrentEvent. The
```
```suggestion
This device produces AC input sent by CurrentEvents. The
```

Reasoning: AC already has a 'current' in it, cf. [PIN number, LCD display, ...](https://en.wikipedia.org/wiki/RAS_syndrome)

jougs commented on models/bernoulli_connection.h:
```diff
@@ -58,8 +58,8 @@ Parameters
 
-Transmits
-+++++++++
+Sends
```
This needs to stay `Transmits`. Also in other files ending in `_connection.h`.

The logic is that neurons have the (single) event type they will send during the simulation under `Sends` and all event types they are able to handle under `Receives`. As synapses/connections only relay the events from the sender to the receiver, the have everything they can relay under `Transmits`.

jougs commented on models/dc_generator.h:
```diff
@@ -38,15 +38,20 @@
 +++++++++++
 
-The DC-Generator provides a constant DC Input to the connected node. The unit of the current is pA.
+The dc_generator provides a constant DC Input to the connected node. The unit of the current is pA.
```
```suggestion
The dc_generator provides a constant DC input to the connected node. The unit of the current is pA.
```

jougs reviewed (changes requested):
Many thanks! This looks mostly good to me :-)

For the future, could you please @mention me (by writing `@jougs` in the description of the PR or a comment somewhere) when you're doing these kinds of things? That way I'd be alerted by email when something is ready for me to look at without any need to communicate that via other means. Thanks!

sarakonradi committed:
Change AC current to AC input

sarakonradi committed:
Change DC Input to DC input

sarakonradi committed:
Retain Transmits in _connection.h files

sarakonradi commented:
@jougs Thanks for your suggestions! 

I added the `AC input`, `DC input` and `Transmits` changes. 

RAS syndrome can be painful. Hence, I removed the repetitions.

jougs commented:
Great!

Can you please again pull `jessica-mitchell:update_models_rst_format` into your branch and fix the conflicts? From my side this would then be good to go.

sarakonradi committed:
Fix ac_generator, dc_generator and volume_transmitter files

sarakonradi committed:
Merge branch 'update_models_rst_format' into update_models_rst_format

jougs commented:
@jessica-mitchell: could you please merge this? I'm fine with the changes now.
@sarakonradi: Merging this one into the branch nest/nest-simulator#1374 is based off should make Travis happy.
